### DISCUSSION TITLE
Update Phoniebox (from 1.1.9 to 2.2)

### DISCUSSION THREAD

**Discussion by jtritter** [MiczFlor/RPi-Jukebox-RFID#1258] [Q&A] [answered]
Hi there,

is there any way to update my from release 1.1.9 to 2.2 without doing a complete new install? Something like an update-script? I have a lot of cards and mp3s on this boxes so creating everything new wouldn't funny. BTW my boxes run Raspbian stretch - do I need an update to buster, too?

Yours
Jürgen

**Comment by s-martin** [comment, marked as answer]
Phoniebox doesn't have a very good upgrade path. So if your box is working and you don't need any of the new features I would suggest not upgrade.

However a few thoughts:
- Stretch support is about to be dropped, so I would also upgrade to Buster; frankly I'm not sure, if the Stretch install script still works
- My suggestion would be to copy/backup the whole current installation directory RPi-Jukebox-RFID, so your music, settings and RFID cards don't get lost
- if you use GPIO be advised we have a new component [GPIO control](https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/develop/components/gpio_control), so you need to manually migrate your GPIO settings after the installation of 2.2 from `gpio_buttons.py` to `settings/gpio_settings.ini`
- After the new installation you should copy your music, playlist, cards from the old saved directory to the new installation

**Reply by jtritter** [reply]
Thank you for the quick reply. I just thought about an update of my 3 boxes due to a new major release of Phoniebox and the "buster-thing". As these 3 boxes sync each other copying the music- and card-files will be no problem. I just wanted to avoid installing 3 boxes completly new while there is an easy to use update-script that I just didn't find...

Yours
Jürgen
